src/lib/core/WeaveFabricState.cpp

```diff
@@ -602,6 +590,294 @@ WEAVE_ERROR WeaveFabricState::GetSharedSessionEndNodeIds(const WeaveSessionKey *
     return err;
 }
 
+void WeaveFabricState::RemoveSharedSessionEndNodes(const WeaveSessionKey *sessionKey)
+{
+    if (sessionKey->IsSharedSession())
+    {
+        SharedSessionEndNode *endNode = SharedSessionsNodes;
+
+        // Clear all information about shared session end nodes.
+        for (int i = 0; i < WEAVE_CONFIG_MAX_SHARED_SESSIONS_END_NODES; i++, endNode++)
+        {
+            if (endNode->SessionKey == sessionKey)
+            {
+                memset((uint8_t *)endNode, 0, sizeof(SharedSessionEndNode));
+            }
+        }
+    }
+}
+
+/**
+ * Suspend and serialize the state of an active Weave Security Session.
+ *
+ */
+WEAVE_ERROR WeaveFabricState::SuspendSession(uint16_t keyId, uint64_t peerNodeId, uint8_t * buf, uint16_t bufSize, uint16_t & serializedSessionLen)
+{
```

Comment:
Do you think this method should try to suspend a session if it is not a shared one?

Reply:
I don't see any reason not to.

Reply:
I guess the rationale was to keep the method agnostic of the session type?

Reply:
Mostly, yes.  By design, it only supports non-connection oriented CASE sessions.  However I didn't see any reason to block its use, e.g., for a device-to-device CASE session.

Reply:
Thanks. Overall LGTM.